doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+================================
+Making your first ARMI-based App
+================================
+
+In this tutorial we will build a nuclear analysis application that runs (dummy) neutron
+flux and thermal/hydraulics calculations. Real applications that do real analysis can be
+modeled after this starting point.
+
+We'll assume you have the :doc:`ARMI Framework installed </user/user_install>` already.
+You can make sure it is ready by running the following command in a shell prompt::
+
+    (armi) $ python -c "import armi;armi.configure()"
+
+You should see an ARMI splash-screen and an ARMI version print out. If you do, you are ready
+to proceed.
+
+.. tip:: If you are having trouble getting it installed, see :ref:`getting-help`. You may
+    need to ensure your ``PYTHON_PATH`` variable includes the armi installation directory.
+
+.. note:: This tutorial is a companion for the :doc:`/developer/making_armi_based_apps`
+    developer documentation.
+
+Starting a new app
+==================
+ARMI-based applications can take on many forms, depending on your workflow. Examples may include:
+
+* Application and plugins together under one folder
+* Application in one folder, plugins in separate ones
+
+For simplicity, we will build an application that contains one plugin that runs
+neutronics and thermal hydraulics in one folder. This architecture will be a good starting
+point for many projects, and can always be separated if needed.
+
+From the command line, ``cd`` into a new directory where you'd like to store your
+application code. Make a folder structure that works as a `normal Python package
+<https://packaging.python.org/tutorials/packaging-projects/>`_, and create some empty
+files for us to fill in, like this::
+
+    my_package/
```

Comment:
`my_armi_project` might be better here. As it is, `my_package` isn't actually a package (no `__init__.py`). If we are going to wade into documenting Python (if however shallow), we should be pedantic.

Reply:
good point. done.